crates/miden-objects/src/asset/vault/partial.rs

```diff
@@ -109,7 +109,8 @@ impl PartialVault {
     /// - the vault key of the proven asset does not match the vault key derived from the asset.
     /// - the new root after the insertion of the leaf and the path does not match the existing root
     ///   (except when the first leaf is added).
-    pub fn add(&mut self, proof: SmtProof) -> Result<(), PartialAssetVaultError> {
+    pub fn add(&mut self, witness: AssetWitness) -> Result<(), PartialAssetVaultError> {
+        let proof = SmtProof::from(witness);
         Self::validate_entries(proof.leaf().entries())?;
```

Comment:
There is no need for validation anymore because asset witness guarantees what we've previously validated. We should also update docs accordingly, so overall:

```rust
/// Adds an [`AssetWitness`] to this [`PartialVault`].
///
/// # Errors
///
/// Returns an error if:
/// - the new root after the insertion of the leaf and the path does not match the existing root
///   (except when the first leaf is added).
pub fn add(&mut self, witness: AssetWitness) -> Result<(), PartialAssetVaultError> {
    let proof = SmtProof::from(witness);
    self.partial_smt
        .add_proof(proof)
        .map_err(PartialAssetVaultError::FailedToAddProof)
}
```